cookbook/python-agent/nodes.py

```
@@ -62,47 +62,44 @@ async def exec(self, inputs):
+
+    async def post(self, memory: Memory, prep_res, exec_res):
         """Save the decision and determine the next step in the flow."""
         # If LLM decided to search, save the search query
         if exec_res["action"] == "search":
-            shared["search_query"] = exec_res["search_query"]
+            memory.search_query = exec_res["search_query"]
             print(f"🔍 Agent decided to search for: {exec_res['search_query']}")
         else:
-            shared["context"] = exec_res["answer"] #save the context if LLM gives the answer without searching.
+            memory.context = exec_res["answer"] #save the context if LLM gives the answer without searching.
             print(f"💡 Agent decided to answer the question")
```

Ruff (0.8.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)